core/clients/key_flow_token_refresh.go

```diff
@@ -0,0 +1,126 @@
+package clients
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/golang-jwt/jwt/v5"
+
+	"github.com/stackitcloud/stackit-sdk-go/core/oapierror"
+)
+
+var (
+	defaultTimeStartBeforeTokenExpiration = 30 * time.Minute
+	defaultTimeBetweenContextCheck        = time.Second
+	defaultTimeBetweenTries               = 5 * time.Minute
+)
+
+// Continuously refreshes the token of a key flow, retrying if the token API returns 5xx errrors. Writes to stderr when it terminates.
+//
+// To terminate this routine, close the context in keyFlow.config.TokenRefreshInBackgroundContext.
+func refreshToken(keyflow *KeyFlow) {
+	refresher := &tokenRefresher{
+		keyFlow:                        keyflow,
+		timeStartBeforeTokenExpiration: defaultTimeStartBeforeTokenExpiration,
+		timeBetweenContextCheck:        defaultTimeBetweenContextCheck,
+		timeBetweenTries:               defaultTimeBetweenTries,
+	}
+	err := refresher.refreshToken()
+	fmt.Fprintf(os.Stderr, "Token refreshing terminated: %v", err)
+}
+
+type tokenRefresher struct {
+	keyFlow *KeyFlow
+	// Token refresh tries start at [Access token expiration timestamp] - [This duration]
+	timeStartBeforeTokenExpiration time.Duration
+	timeBetweenContextCheck        time.Duration
+	timeBetweenTries               time.Duration
+}
+
+// Continuously refreshes the token of a key flow, retrying if the token API returns 5xx errrors. Always returns with a non-nil error.
+//
+// To terminate this routine, close the context in refresher.keyFlow.config.TokenRefreshInBackgroundContext.
+func (refresher *tokenRefresher) refreshToken() error {
```

Comment:
maybe just call this `refresh`

there's already too many `refresh` and `token` going around, plus having two methods with the same name is strange

Reply:
`refresh` really doesn't say what this does
Renamed to `continuousTokenRefresh` and renamed the filed
If you have better names, please let me know